include/boost/geometry/util/for_each_with_index.hpp

```diff
@@ -30,6 +30,17 @@ inline void for_each_with_index(Container const& container, Function func)
     }
 }
 
+template <typename Container, typename Function>
+inline void for_each_with_index(Container& container, Function func)
+{
+    std::size_t index = 0;
+    for (auto it = std::begin(container); it != std::end(container); ++it, ++index)
```

Comment:
Technically the type of index should be taken from Container/Range, e.g.: `boost::size_type<Container>::type` but I guess in practice `std::size_t` will be ok.

Are we sure that this utility will not be used for geometries adapted to Boost.Range Concept?
Would it make sense to use `boost::begin` and `boost::end` to be safe?

Reply:
:heavy_check_mark: fixed. While fixing it, I now also use Boost Range's size type